lib/vtls/openssl.c

```diff
@@ -65,6 +65,9 @@
 #include <openssl/bio.h>
 #include <openssl/buffer.h>
 #include <openssl/pkcs12.h>
+#ifdef USE_ESNI
```

Comment:
`USE_ESNI` might in the future be set for any TLS backend that support sit, so don't assume that this means OpenSSL support!

Reply:
It seemed redundant to have a more elaborate test here, as code in this file (apart from the one line `#include "curl_setup.h"`) is subject to `#ifdef USE_OPENSSL`. Would `#ifdef HAVE_OPENSSL_ESNI_H` be preferable?